### DISCUSSION TITLE
[(weird)ERROR]: `belonging_to`, perhaps you need to implement it: candidate #1: `diesel::BelongingToDsl`

### DISCUSSION THREAD

**Discussion by jjangga0214** [diesel-rs/diesel#3830] [Q&A]
Hi!

I have compiler error for `belonging_to`.
I think the project is configured well.
But I don't know why this is happening.

```rs
// src/schema.rs
use diesel::sql_types::*;
use diesel::{allow_tables_to_appear_in_same_query, joinable, table};

table! {
    user (id) {
        id -> Uuid
    }
}

table! {
    count (id) {
        id -> Uuid,
        user_id -> Uuid,
    }
}

joinable!(count -> user (user_id));

allow_tables_to_appear_in_same_query!(user, count);
```

```rs
// src/models/count/generated.rs
#[derive(Debug, Serialize, Deserialize, Clone, Queryable, Insertable, AsChangeset, Identifiable, Associations, Selectable)]
#[diesel(table_name=count, primary_key(id), belongs_to(User, foreign_key=user_id))]
pub struct Count {
    pub id: uuid::Uuid,
    pub user_id: uuid::Uuid
}
```

<img width="845" alt="스크린샷 2023-10-17 오전 1 43 49" src="https://github.com/Wulf/dsync/assets/28584151/530cb104-cc0c-4d90-8e1c-0f34a52eb52a">

Thank you :)

**Comment by weiznich** [comment]
It's not possible to answer this without knowing the relevant context. Please at at least the following details to your questions:

* Information about your diesel version
* Information about your rustc version
* Information about your operating system
* Information about your database system (which one is used + what version)
* The literal error message as emitted by rustc
* Relevant parts of your schema, code, … This includes definitions of all types that are not from diesel or rust-stdlib and the relevant types of all involved variables.
